modules/library/src/main/scala/zio/elasticsearch/ElasticAggregationResponse.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.json._
+import zio.json.ast.Json.Obj
+import zio.json.ast._
+import zio.json.internal.RetractReader
+
+sealed trait ElasticAggregationResponse
+
+final case class TermsAggregationResponse(
+  @jsonField("doc_count_error_upper_bound")
+  docErrorCount: Int,
+  @jsonField("sum_other_doc_count")
+  sumOtherDocCount: Int,
+  buckets: List[TermsAggregationBucket]
+) extends ElasticAggregationResponse
+
+object TermsAggregationResponse {
+  implicit val decoder: JsonDecoder[TermsAggregationResponse] = DeriveJsonDecoder.gen[TermsAggregationResponse]
+}
+
+sealed trait ElasticAggregationBucket
+
+final case class TermsAggregationBucket(
+  key: String,
+  @jsonField("doc_count")
+  docCount: Int,
+  subAggregations: Option[Map[String, ElasticAggregationResponse]] = None
+) extends ElasticAggregationBucket
+
+object TermsAggregationBucket {
+  implicit val decoder: JsonDecoder[TermsAggregationBucket] = (trace: List[JsonError], in: RetractReader) => {
+    val json = JsonDecoder[Json].unsafeDecode(trace, in)
+
+    json match {
+      case Obj(fields) =>
+        val allFields = fields.flatMap { case (field, data) =>
+          field match {
+            case "key" =>
+              Some(field -> data.toString.replaceAll("\"", ""))
+            case "doc_count" =>
+              data.as[Int].toOption.map(field -> _)
+            case _ =>
+              val objFields = Obj.decoder.decodeJson(data.toString).toOption.get.fields.toMap
+
+              field match {
+                case str if str.contains("terms") =>
```

Comment:
Should we use something more restrictive than `contains`? Could this be a problem if someone defines a custom aggregation like `mytermsagg`?